core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -787,7 +791,9 @@ private[spark] class ExecutorAllocationManager(
     /**
      * The number of tasks currently running across all stages.
      */
-    def totalRunningTasks(): Int = numRunningTasks
+    def totalRunningTasks(): Int = {
+      stageIdToNumRunningTask.values.sum
```

Comment:
This needs to be inside `allocationManager.synchronized`, no?

Reply:
Nevermind, this is called from a synchronized context. Except in your unit tests, that is (which call the private`totalRunningTasks` you added to the manager).

Reply:
It'd be nice to make the other method calling this synchronized, just to be paranoid.